cimgen/languages/modernpython/lang_pack.py

```diff
@@ -76,23 +79,100 @@ def _get_type_and_default(text, render) -> tuple[str, str]:
+        result += "from ." + val + " import " + val + "\n"
+    return result + "\n"
+
+
+def _set_datatype_attributes(attributes) -> dict:
+    datatype_attributes = {}
+    datatype_attributes["python_type"] = "None"
+    datatype_attributes["unit"] = "UnitSymbol.none"
+    datatype_attributes["multiplier"] = "UnitMultiplier.none"
+
+    for attribute in attributes:
+        if (
+            "about" in attribute
+            and attribute["about"]
+            and "value" in attribute["about"]
+            and "attribute_class" in attribute
+        ):
+            datatype_attributes["python_type"] = _primitive_to_data_type(attribute["attribute_class"])
+        if (
+            "about" in attribute
+            and attribute["about"]
+            and "multiplier" in attribute["about"]
+            and "isFixed" in attribute
+        ):
+            datatype_attributes["multiplier"] = "UnitMultiplier." + attribute["isFixed"]
+        if "about" in attribute and attribute["about"] and "unit" in attribute["about"] and "isFixed" in attribute:
+            datatype_attributes["unit"] = "UnitSymbol." + attribute["isFixed"]
+    return datatype_attributes
+
+
 def run_template(output_path, class_details):
-    if class_details["is_a_primitive_class"] or class_details["is_a_datatype_class"]:
-        return
-    for template_info in template_files:
-        resource_file = Path(
-            os.path.join(
-                output_path,
-                "resources",
-                class_details["class_name"] + template_info["ext"],
-            )
-        )
-        if not resource_file.exists():
-            if not (parent := resource_file.parent).exists():
-                parent.mkdir()
+    if class_details["is_a_primitive_class"]:
+        # Primitives are never used in the in memory representation but only for
+        # the schema
+        template = primitive_template_file
+        class_details["python_type"] = _primitive_to_data_type(class_details["class_name"])
+    elif class_details["is_a_datatype_class"]:
+        # Datatypes based on primitives are never used in the in memory
+        # representation but only for the schema
+        template = datatype_template_file
+        class_details.update(_set_datatype_attributes(class_details["attributes"]))
+    elif class_details["is_an_enum_class"]:
+        template = enum_template_file
+    else:
+        template = class_template_file
         class_details["setDefault"] = _set_default
         class_details["setType"] = _set_type
-        _write_templated_file(resource_file, class_details, template_info["filename"])
+        class_details["setImports"] = _set_imports(class_details["attributes"])
+    resource_file = _create_file(output_path, class_details, template)
+    _write_templated_file(resource_file, class_details, template["filename"])
+
+
+def _create_file(output_path, class_details, template) -> str:
+    resource_file = Path(
+        os.path.join(
```

Comment:
as `pathlib` is already used, maybe use [PurePath.joinpath()](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.joinpath)?